examples/ft_transformer_text.py

```diff
@@ -51,21 +52,147 @@ def __call__(self, sentences: List[str]) -> Tensor:
 parser.add_argument('--lr', type=float, default=0.0001)
 parser.add_argument('--epochs', type=int, default=100)
 parser.add_argument('--seed', type=int, default=0)
+parser.add_argument('--finetune', action='store_true')
+parser.add_argument('--lora', action='store_true')
+parser.add_argument(
+    '--model', type=str, default='distilbert-base-uncased', choices=[
+        'distilbert-base-uncased',
+        'sentence-transformers/all-distilroberta-v1',
+    ])
+parser.add_argument('--pooling', type=str, default='mean',
+                    choices=['mean', 'cls'])
 args = parser.parse_args()
 
+if args.lora and not args.finetune:
+    raise ValueError('Please also specify finetune when '
+                     'choosing LoRA finetuning.')
+
+
+class PretrainedTextEncoder:
+    def __init__(self, model: str, pooling: str, device: torch.device):
+        self.tokenizer = AutoTokenizer.from_pretrained(model)
+        self.model = AutoModel.from_pretrained(model).to(device)
+        self.device = device
+        self.pooling = pooling
+
+    def __call__(self, sentences: List[str]) -> Tensor:
+        inputs = self.tokenizer(sentences, truncation=True,
+                                padding='max_length', return_tensors='pt')
+        for key in inputs:
+            if isinstance(inputs[key], Tensor):
+                inputs[key] = inputs[key].to(self.device)
+        out = self.model(**inputs)
+        mask = inputs['attention_mask']
+        if self.pooling == 'mean':
+            return mean_pooling(out.last_hidden_state.detach(),
+                                mask).squeeze(1).cpu()
+        elif self.pooling == 'cls':
+            return out.last_hidden_state[:, 0, :].detach().cpu()
+        else:
+            raise ValueError(f'{self.pooling} is not supported.')
+
+
+class TextTokenizer:
+    def __init__(self, model: str):
+        self.tokenizer = AutoTokenizer.from_pretrained(model)
+
+    def __call__(self, sentences: List[str]) -> List[Dict[str, Tensor]]:
+        res = []
+        for s in sentences:
+            inputs = self.tokenizer(s, truncation=True, padding=False,
+                                    return_tensors='pt')
+            res.append(inputs)
+        return res
+
+
+class TextEncoder(torch.nn.Module):
+    def __init__(self, model: str, pooling: str = 'mean', lora: bool = False):
```

Comment:
Can we have some doc-string for `TextEncoder`? especially, what do you mean by `model`? What kind of string do you expect?